extbench/runoctane.py

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env python2.7
+
+import csv, os, sys, time
+from decimal import Decimal
+from krun.platform import detect_platform
+from krun.util import run_shell_cmd_bench
+
+WARMUP_DIR = os.path.realpath(os.path.dirname(os.path.dirname(__file__)))
+
+JAVASCRIPT_VMS = {
+    "v8": "sh -c 'cd %s/extbench/octane && LD_LIBRARY_PATH=%s/krun/libkrun %s/work/v8/out/native/d8 run.js'"
+          % (WARMUP_DIR, WARMUP_DIR, WARMUP_DIR)
+}
+
+ITERATIONS = 2000
+PROCESSES = 10
+
+def main():
+    platform = detect_platform(None, None)
+    for jsvm_name, jsvm_cmd in JAVASCRIPT_VMS.items():
+        with open("octane.%s.results" % jsvm_name, 'wb') as csvf:
+            sys.stdout.write("%s:" % jsvm_name)
+            writer = csv.writer(csvf)
+            writer.writerow(['processnum', 'benchmark'] + range(ITERATIONS))
+            for process in range(PROCESSES):
+                sys.stdout.write(" %s" % str(process))
+                sys.stdout.flush()
+                # Flush the CSV writing, and then give the OS some time
+                # to write stuff out to disk before running the next process
+                # execution.
+                csvf.flush()
+                os.fsync(csvf.fileno())
+                time.sleep(3)
+
+                stdout, stderr, rc = run_shell_cmd_bench(jsvm_cmd, platform)
+                if rc != 0:
+                    sys.stderr.write(stderr)
+                    sys.exit(rc)
+                times = None
```

Comment:
If I were to nitpick, I'd add a comment to each of the cases here. I think one consumes the next measurement, and the other flushes a whole execution's worth of measurements to disk. Am I right?